app/src/main/java/org/openobservatory/ooniprobe/model/database/Result.java

```diff
@@ -58,7 +58,7 @@ public static Result getLastResult(String test_group_name) {
 
 	public static void deleteAll(Context c) {
 		try {
-			FileUtils.cleanDirectory(Measurement.getMeasurementDir(c));
+			FileUtils.cleanDirectory(Measurement.createMeasurementDir(c));
```

Comment:
I didn't actually realise in all the places in which this is being called. I think that since it's being used also in situations where it doesn't make sense for it to be called this way, then I suggest you do as I originally suggested and split the creation of the directory from getting the pathname.

Reply:
we need to talk about that

Reply:
in this way i don't care if the folder exists or not. i can use abstracting from that. what do you mean in particular ?

Reply:
I mean that if you read that code above it doesn't make any sense that you are `createMeasurementDir` then to delete it. So the function name should be reflected in that no?

Reply:
So maybecreate or back to get

Reply:
As I have already written in a couple of places

> I suggest you do as I originally suggested and split the creation of the directory from getting the pathname

Reply:
I use this compact way to abstract from know if folder exists or not in every place I use it. How can I do same?

Reply:
My suggestion is create the folder once and then use it whenever you need to access it.

If while you are deleting the folder you notice that it doesn't exist and an exception happens, that's probably a good thing, no?

Reply:
Yes, I was thinking that filesystem is not under exclusively of app (is not simple for user to access that folder) so all interaction assume that fs can be modified

Reply:
https://github.com/ooni/probe-android/pull/217#discussion_r277803255

Reply:
`./hellais explain --verbose`

1. Rename `createMeasurementDir` to `getMeasurementDir`
2. Remove the `dir.mkdirs();` call from inside `getMeasurementDir` and place it inside of all the places in which you may be creating the directory for the first time, after you have obtained the `measuremetnDir`.
Example:
https://github.com/ooni/probe-android/blob/b8f1799d5cb74d89652b5d8bd927b9eee54372b4/app/src/main/java/org/openobservatory/ooniprobe/test/test/AbstractTest.java#L109
https://github.com/ooni/probe-android/blob/b8f1799d5cb74d89652b5d8bd927b9eee54372b4/app/src/main/java/org/openobservatory/ooniprobe/test/test/AbstractTest.java#L91
It's possibly cleaner to do that before you even go into the switch case imho though